### ISSUE TITLE
Iteration Plan: Autumn/Winter 2020

### ISSUE THREAD

**Issue by johnbillion** [johnbillion/query-monitor#539]
This plan captures the current high level goals for Query Monitor. The list might change over time and the next plan will start in Spring 2021.

This plan isn't tied to any particular releases. Some planned items need an issue to be opened as they've been moved here from my notes app. Other bug fixes will be included in releases but aren't necessarily listed here.

## Priority Fixes

* [ ] Make `QM_Backtrace` more resilient, reliable, add tests - #564
  - [ ] #495
  - [ ] #494 
  - [ ] #461 
  - [ ] #433 
  - [ ] #340 
* [x] Remove warning when certification verification is disabled for local URLs #537 

## New Panels

* [ ] `WP_Error` object tracking #559
* [ ] Object cache statistics standard #518

## General Features

* [x] Always show the Logs panel, with a link to help docs
* [x] Remove QM by default from places where it technically should show but practically isn't useful
  - [x] #355
  - [x] #335
* [ ] 404 helper #342
* [ ] ~~Ability for code partway through a request to prevent QM's output~~ This is already possible with the `qm/dispatch/html` filter. Need a way to cease the data collection and processing though.
* [ ] Add time on running hooks #260 (stretch goal?)

## Performance and Dependencies

* [x] Prepare for jQuery updates #473
* [ ] Limit the number of HTTP headers for PHP errors that occur during Ajax, REST API, etc #275 

## WordPress Core Tasks

* [X] Add `WP_Error` tracking [#WP40568](https://core.trac.wordpress.org/ticket/40568)
* [ ] Add an action when a rewrite rule is added or removed, including endpoints etc

## Meta

* [x] GitHub action for building a tag so the Composer package can be un-abandoned #551 
* [x] Switch to GitHub action for semver milestone management (currently Probot) #552
* [ ] General website updates, TBD
* [ ] Publish info about [sponsorship](https://github.com/sponsors/johnbillion)

## Ongoing

Long (long) term work which is in progress but very unlikely to ship soon

* [ ] Implement client side rendering 🚀  #483 , the God-ticket which will ultimately fix or unblock:
  - #469 
  - #456 
  - #379 
  - #354 
  - #291 
  - #223 
  - #174 
  - #173 
  - #39 
* [ ] Hook inspection
* [ ] Component insights
* [ ] Flow explorer
* [ ] Document all the actions and filters

---

Thanks to [microsoft/vscode](https://github.com/microsoft/vscode), [Kong/insomnia](https://github.com/Kong/insomnia), and others for the idea of publishing iteration plans.

**Comment by johnbillion** [comment]
Opened #564 for improvements to stack trace filtering and component detection

**Comment by johnbillion** [comment]
#594
